core/pubnub_grant_token_api.c

```diff
@@ -99,40 +99,6 @@ pubnub_chamebl_t pubnub_get_grant_token(pubnub_t* pb)
     return result;
 }
 
-char* pubnub_parse_token(pubnub_t* pb, char const* token){
-    char * rawToken = strdup(token);
-    replace_char((char*)rawToken, '_', '/');
-    replace_char((char*)rawToken, '-', '+');
-
-    pubnub_bymebl_t decoded;
-    decoded = pbbase64_decode_alloc_std_str(rawToken);
-    #if PUBNUB_LOG_LEVEL >= PUBNUB_LOG_LEVEL_DEBUG
-    PUBNUB_LOG_DEBUG("\nbytes after decoding base64 string = [");
-    for (size_t i = 0; i < decoded.size; i++) {
-        PUBNUB_LOG_DEBUG("%d ", decoded.ptr[i]);
-    }
-    PUBNUB_LOG_DEBUG("]\n");
-    #endif
-
-    size_t length;
-    uint8_t *buf = decoded.ptr;
-    length = decoded.size;
-
-    CborParser parser;
-    CborValue it;
-    
-    char * json_result = (char*)malloc(5*(strlen(rawToken)/4));
-    sprintf(json_result, "%s", "");
-    CborError err = cbor_parser_init(buf, length, 0, &parser, &it);
-    if (!err){
-        data_recursion(&it, 1, json_result);
```

Comment:
Other way to fix this without moving a large chunk of code (making diff harder to review) is adding a function definition for `data_recursion` on line 101 or higher. Usually, the function definitions appear in header files, but then you may have scoping issues (like accidentally exposing internal functions and types to external callers) and trigger other errors for other language quirks like not being allowed to define a function more than once.

There are workarounds for these issues, as well, but if `data_recursion` is only referenced in this file, it's fine to define it in this file.